src/V3Assert.cpp

```diff
@@ -505,6 +505,24 @@ class AssertVisitor final : public VNVisitor {
         } else if (nodep->displayType() == VDisplayType::DT_MONITOR) {
             nodep->displayType(VDisplayType::DT_DISPLAY);
             const auto fl = nodep->fileline();
+            AstNode* Monexp =  nodep->fmtp()->exprsp(); 
+            AstSenItem* MonSenItem = NULL;
+            int monsenflag =1;
+            while(Monexp != NULL)
+            {
+                if(Monexp->type() == VNType::atVarRef)
+                 {  
+                    if(monsenflag) {
+                        MonSenItem= new AstSenItem(fl, VEdgeType::ET_CHANGED, (AstNodeExpr*)(new AstVarRef{fl, ((AstNodeVarRef*) Monexp)->varp(), VAccess::READ}));
+                        monsenflag = 0;
+                    } 
+                    else {
+                        MonSenItem->addNext(new AstSenItem(fl, VEdgeType::ET_CHANGED, (AstNodeExpr*)(new AstVarRef{fl, ((AstNodeVarRef*) Monexp)->varp(), VAccess::READ})));
+                    }
```

Comment:
Create it once, then have the if statement after to decide to set monSenItemp either directly or with an addNext.